src/lib.rs

```diff
@@ -0,0 +1,409 @@
+        // migrate(pool).await?;
+
+        sqlx::migrate!("db/migrations").run(pool).await.unwrap();
+
+        let mut changeset = ChangeSet::default();
+
+        let mut tx = pool.begin().await.unwrap();
+
+        // Fetch wallet data
+        let row = sqlx::query_as::<_, WalletRow>(
+            "SELECT n.name as network, k.descriptor, k.last_revealed 
+             FROM network n 
+             JOIN keychain k ON n.wallet_name = k.wallet_name 
+             LIMIT 1",
+        )
+        .fetch_optional(&mut *tx)
+        // .fetch_optional(&pool)
+        .await
+        .unwrap();
+
+        if let Some(WalletRow {
+            network,
+            descriptor,
+            last_revealed,
+        }) = row
+        {
+            let (descriptor, _) =
+                ExtendedDescriptor::parse_descriptor(&Secp256k1::new(), &descriptor).unwrap();
+            let did = descriptor.descriptor_id();
+            changeset.descriptor = Some(descriptor);
+            changeset.network = Some(Network::from_str(&network).expect("must parse"));
+            changeset.indexer = keychain_txout::ChangeSet {
+                last_revealed: [(did, last_revealed as u32)].into(),
+            };
+        }
+
+        changeset.tx_graph = tx_graph_changeset_from_postgres(&mut tx).await.unwrap();
+        changeset.local_chain = local_chain_changeset_from_postgres(&mut tx).await.unwrap();
+        // Note: keychain_txout ChangeSet is already set above
+
+        tx.commit().await.unwrap();
+
+        Ok(changeset)
+    }
+
+    async fn write(&self, changeset: &ChangeSet, wallet_name: String) -> Result<(), Error> {
+        if changeset.is_empty() {
+            return Ok(());
+        }
+
+        let pool = &self.pool;
+        let mut tx = pool.begin().await.unwrap();
+
+        // let network = self.network;
+
+        // let descriptor = changeset.clone().descriptor.unwrap();
+        // let change_descriptor = changeset.clone().change_descriptor;
+        //
+        // let wallet_name = bdk_wallet::wallet_name_from_descriptor(descriptor,change_descriptor , network,&Secp256k1::new()).unwrap();
+
+        if let Some(ref descriptor) = changeset.descriptor {
+            insert_descriptor(&mut tx, &wallet_name, descriptor)
+                .await
+                .unwrap();
+        }
+        if let Some(network) = changeset.network {
+            insert_network(&mut tx, &wallet_name, network)
+                .await
+                .unwrap();
+        }
+        if let Some(last_revealed) = changeset.indexer.last_revealed.values().next() {
+            update_last_revealed(&mut tx, &wallet_name, *last_revealed)
+                .await
+                .unwrap();
+        }
+
+        tx_graph_changeset_persist_to_postgres(&mut tx, &wallet_name, &changeset.tx_graph)
+            .await
+            .unwrap();
+        local_chain_changeset_persist_to_postgres(&mut tx, &wallet_name, &changeset.local_chain)
+            .await
+            .unwrap();
```

Comment:
Since the table `anchor_tx` references `block` I think we should persist the local chain blocks before the tx-graph changes